manifests/install.pp

```diff
@@ -1,29 +1,46 @@
 # == Class vault::install
 #
 class vault::install {
+
   $vault_bin = "${::vault::bin_dir}/vault"
 
   case $::vault::install_method {
-      'archive': {
-        if $::vault::manage_download_dir {
-          file { $::vault::download_dir:
-            ensure => directory,
-          }
+    'archive': {
+      if $::vault::manage_download_dir {
+        file { $::vault::download_dir:
+          ensure => directory,
         }
+      }
 
-        archive { "${::vault::download_dir}/${::vault::download_filename}":
-          ensure       => present,
-          extract      => true,
-          extract_path => $::vault::bin_dir,
-          source       => $::vault::real_download_url,
-          cleanup      => true,
-          creates      => $vault_bin,
-          before       => File['vault_binary'],
-        }
+      $_manage_file_capabilities = true
+      $_vault_versioned_bin = "/opt/vault-${::vault::version}/vault"
+
+      file { "/opt/vault-${::vault::version}":
+        ensure => directory,
+        owner  => 'root',
+        group  => 'root',
+        mode   => '0755',
+      }
 
-        $_manage_file_capabilities = true
+      archive { "${::vault::download_dir}/${::vault::download_filename}":
+        ensure       => present,
+        extract      => true,
+        extract_path => "/opt/vault-${::vault::version}",
+        source       => $::vault::real_download_url,
+        cleanup      => true,
+        creates      => $_vault_versioned_bin,
+        before       => File['vault_binary'],
+        notify       => Exec['install_versioned_vault'],
       }
 
+      exec { 'install_versioned_vault':
+        command     => "/bin/cp -f ${_vault_versioned_bin} ${vault_bin}",
```

Comment:
Any reason to do a `cp` instead of a symlink? This approach was discussed in #63 but I can't recall the specifics of if it was/wasn't feasible.

Reply:
I tried with a symlink. But the issue is that you can't set the file_capabilities on a symlink. Keeping to the symlink path meant changing the code in other places. This was the easiest way to introduce this functionality.

Reply:
@rgevaert symlinks are pass-through, so whatever they link to is the mode in play.

E.g., if you symlink `${_vault_versioned_bin}` → `${vault_bin}`, whatever `mode` is assigned to `${vault_bin}` will be the mode used.

So this is probably cleaner done as a symlink:

```puppet
file { $vault_bin:
  ensure => link,
  target => $vault_versioned_bin,
  notify => Class['vault::service'],
}
```

And let whatever mode the actual bin has be the mode in play. If it needs to be set, I believe you’ll want to do that in the `Archive["${::vault::download_dir}/${::vault::download_filename}"]` resource above.

Reply:
TBH  I tried that first but had issues like mentioned before. If someone else has time to pick it up, please do.

Reply:
Using a symlink will throw an error when setting `mlock` on the `$vault_bin`, which is done at https://github.com/jsok/puppet-vault/blob/97dd7941f078b6da438ad41600bcf7563533bcc0/manifests/install.pp#L66. A way to fix this would be to reference `$_vault_versioned_bin` instead.